src/handler/http/request/folders.rs

```diff
@@ -99,42 +133,29 @@ pub async fn update_folder(
         ("org_id" = String, Path, description = "Organization name"),
     ),
     responses(
-        (status = StatusCode::OK, body = FolderList),
+        (status = StatusCode::OK, body = ListFoldersResponseBody),
     ),
 )]
 #[get("/{org_id}/folders")]
-pub async fn list_folders(
-    path: web::Path<String>,
-    _req: HttpRequest,
-) -> Result<HttpResponse, Error> {
+#[allow(unused_variables)]
+pub async fn list_folders(path: web::Path<String>, req: HttpRequest) -> impl Responder {
     let org_id = path.into_inner();
 
-    let mut _permitted = None;
-    // Get List of allowed objects
+    #[cfg(not(feature = "enterprise"))]
+    let user_id = None;
+
     #[cfg(feature = "enterprise")]
-    {
-        let user_id = _req.headers().get("user_id").unwrap();
-        match crate::handler::http::auth::validator::list_objects_for_user(
-            &org_id,
-            user_id.to_str().unwrap(),
-            "GET",
-            "dfolder",
-        )
-        .await
-        {
-            Ok(stream_list) => {
-                _permitted = stream_list;
-            }
-            Err(e) => {
-                return Ok(crate::common::meta::http::HttpResponse::forbidden(
-                    e.to_string(),
-                ));
-            }
```

Comment:
This logic is moved into the Service Layer since. Extracting a `user_id` from the HTTP headers is logic specific to the HTTP API and is left here. However the actual authorization logic performed here seems like something core to the domain logic that we want to perform, independent of and agnostic to the actually REST HTTP API. (For example, if we ever wanted to expose the "List Folders" logic through some other form of API, such as gRPC, GraphQL, or even a CLI... then I'm assuming that we would still want this logic to be part of the core "list folders" logic that executes.)